types/payload.go

```diff
@@ -151,6 +182,11 @@ func (p *Payload) Get(key string) any {
 		}
 	}
 
+	if p.missingFields == nil {
+		p.missingFields = make(map[string]struct{})
+	}
+	p.missingFields[key] = struct{}{}
```

Comment:
I don't think we should implicitly memoize - that has a non-trivial cost. The assumption should be that a field is accessed once only, and if it's going to be more than that, it's the subject of a MemoizeFields call.